server/routes/search.ts

```diff
@@ -11,15 +11,70 @@ const Errors = {
   UnexpectedError: 'Unexpected error',
   QueryMissing: 'Must enter query to begin searching',
   QueryTooShort: 'Query must be at least 4 characters',
+  NoCommunity: 'Title search must be community scoped'
 };
 
-const search = async (models: DB, req: Request, res: Response, next: NextFunction) => {
+const search = async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
   let replacements = {};
 
+  if (req.query.thread_title_only === 'true') {
```

Comment:
Let's try to use an [early-return pattern](https://szymonkrajewski.pl/why-should-you-return-early/) to make this large `if` conditional more readable.

Reply:
Little confused, I think I basically am? This conditional isn't error-catching, it's specifying a different query (only search thread titles).

Reply:
Something like a switch case would be "clearer" here being that there is so much within this `if` conditional.